packages/gestalt/src/Link.js

```diff
@@ -18,6 +18,59 @@ import getRoundingClassName from './getRoundingClassName.js';
 import focusStyles from './Focus.css';
 import useFocusVisible from './useFocusVisible.js';
 import layoutStyles from './Layout.css';
+import Icon from './Icon.js';
+import Box from './Box.js';
+import Text from './Text.js';
+
+type ExternalLinkIcon =
+  | 'none'
+  | 'default'
+  | {|
+      size: $ElementType<React$ElementConfig<typeof Text>, 'size'>,
+      color: $ElementType<React$ElementConfig<typeof Icon>, 'color'>,
+    |};
+
+function ExternalIcon({
+  externalLinkIcon,
+  position,
+}: {|
+  externalLinkIcon: ExternalLinkIcon,
+  position: 'rtl' | 'ltr',
+|}): Node {
+  const externalLinkIconMap = {
+    '100': 12,
+    '200': 14,
+    '300': 16,
+    '400': 20,
+    '500': 28,
+    '600': 36,
+    'sm': 12,
+    'md': 14,
+    'lg': 16,
+  };
+
+  return externalLinkIcon === 'none' ? null : (
+    <div aria-hidden className={classnames(position === 'rtl' ? styles.rtlIcon : styles.ltrIcon)}>
```

Comment:
nit: consider some variables for readability:
```js
const isRtl = position === 'rtl';
const isDefault = externalLinkIcon === 'default';

let iconSize = '300';
let iconColor = 'default';
if (!isDefault) {
  // I'm pretty sure Flow will be ok with this, but it might complain
  iconSize = externalLinkIcon?.size ?? iconSize;
  iconColor = externalLinkIcon?.color ?? iconColor;
}

return (
  <div aria-hidden className={classnames(styles[isRtl ? 'rtlIcon' : 'ltrIcon')}>
      <Box
        marginEnd={isRtl ? 1 : 0}
        marginStart={isRtl ? 0 : 1}
        display="inlineBlock"
      >
        <Icon
          inline
          icon="visit"
          accessibilityLabel=""
          size={externalLinkIconMap[iconSize]}
          color={iconColor}
        />
      </Box>
    </div>
  );
```